vcstool/clients/git.py

```diff
@@ -395,6 +395,86 @@ def status(self, command):
             cmd += ['--untracked-files=no']
         return self._run_command(cmd)
 
+    def validate(self, command):
+        if not command.url:
+            return {
+                'cmd': '',
+                'cwd': self.path,
+                'output': "Repository data lacks the 'url' value",
+                'returncode': 1
+            }
+
+        self._check_executable()
+
+        cmd_ls_remote = [GitClient._executable, 'ls-remote']
+        cmd_ls_remote += ['-q', '--exit-code']
+        cmd_ls_remote += [command.url]
+        result_ls_remote = self._run_command(
+            cmd_ls_remote,
+            retry=command.retry,
+            env={'GIT_TERMINAL_PROMPT': '0'})
+        if result_ls_remote['returncode']:
+            result_ls_remote['output'] = \
+                "Could not contact remote repository '%s': %s" % \
+                (command.url, result_ls_remote['output'])
+            return result_ls_remote
+
+        if command.version:
+            hashes = []
+            refs = []
+            ref_found = False
+            output_lines = result_ls_remote['output'].splitlines()
+
+            for line in output_lines:
+                hash_and_ref = line.split()
+                hashes.append(hash_and_ref[0])
+
+                # ignore pull request refs
+                if hash_and_ref[1].find('refs/pull/') == -1:
```

Comment:
The ref has to actually start with `refs/...`, no? In that case instead of using `find` `startswith` makes more sense.

Same below.